### PR TITLE
Upstream `waf` & `acm`

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Upstreams `waf`
* Upstreams `acm`

## why
* Upstreaming for public use

## references
* N/A

### CODE REVIEW

File: modules/acm/default.auto.tfvars [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file is included by default in terraform plans
+
+enabled = false
+
+name = "acm"
```
Comment (aknysh) [inline modules/acm/default.auto.tfvars:5]: please move `name` to YAML config (e.g. `catalog/acm.yaml` vars).
we had issues in Spacelift when we define vars in many diff places

File: modules/acm/main.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  context = module.this.context
+}
+
+resource "aws_ssm_parameter" "acm_arn" {
+  count = local.enabled ? 1 : 0
+
+  name        = "/acm/${var.domain_name}"
+  value       = module.acm.arn
+  description = "ACM certificate id"
```
Comment (aknysh) [inline modules/acm/main.tf:36]:
```suggestion
  description = format("ACM certificate ARN for '%s' domain", var.domain_name)
```

File: modules/waf/default.auto.tfvars [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# This file is included by default in terraform plans
+
+enabled = false
+
+name = "waf"
```
Comment (aknysh) [inline modules/waf/default.auto.tfvars:5]: please move `name` and `default_action` from here to YAML stack config in the catalog for the component

File: modules/waf/main.tf [hunk 4]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (aknysh) [inline modules/waf/main.tf:1]: remove empty line

File: modules/waf/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,4 @@
+output "acl" {
```
Comment (aknysh) [inline modules/waf/outputs.tf:1]: prob update the name of the output since it outputs the entire an AWS Web Application Firewall, not just ACL

**Review by aknysh — COMMENTED** [review]
please see comments

**Review by nitrocode — DISMISSED** [review]
Looks good to me but let's wait for @aknysh to re-review

File: modules/acm/default.auto.tfvars (hunk 1 above)
Comment (aknysh) [reply modules/acm/default.auto.tfvars]: @Benbentwo  you removed the `name` from the file, but did you add it to YAML config for `acm`?

**Review by aknysh — COMMENTED** [review]
please see comments

File: modules/waf/default.auto.tfvars (hunk 3 above)
Comment (Benbentwo) [reply modules/waf/default.auto.tfvars]: Updated Readme

File: modules/waf/README.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| <a name="output_waf"></a> [waf](#output\_waf) | Information about the created WAF ACL |
+<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+
+## References
+* [cloudposse/terraform-aws-components](https://github.com/cloudposse/terraform-aws-components/tree/master/modules/ecr) - Cloud Posse's upstream component
```
Comment (milldr) [inline modules/waf/README.md:120]: change `ecr` to TODO or `waf` if available

**Review by milldr — APPROVED** [review]

**Pull request merged**